array-manipulation-1/src/compact.ts

```diff
@@ -0,0 +1,18 @@
+/* exported compact */
+function compact(array: any[]): any[] {
+  const goodValues: any[] = [];
+  for (let i = 0; i < array.length; i++) {
+    const value = array[i];
+    if (
+      value !== false &&
+      value !== null &&
+      value !== undefined &&
+      value !== '' &&
+      !Number.isNaN(value) &&
+      value !== 0
+    ) {
```

Comment:
While this may work, there is a better way to check for all of these values. Take a look into `truthy` and `falsy` values and you can make a much cleaner to read condition here!